samza-core/src/main/scala/org/apache/samza/metrics/reporter/MetricsSnapshotReporter.scala

```diff
@@ -70,6 +71,25 @@ class MetricsSnapshotReporter(
   info("got metrics snapshot reporter properties [job name: %s, job id: %s, containerName: %s, version: %s, samzaVersion: %s, host: %s, pollingInterval %s]"
     format (jobName, jobId, containerName, version, samzaVersion, host, pollingInterval))
 
+  val partitionKeyValue = getMetricsReporterStreamPartitionKeyValue(partitionKeyName, jobName, Util.getLocalHost.getHostName)
+  info("Using PartitionKey %s = %s" format(partitionKeyName, partitionKeyValue))
+
+  /**
+    * Returns the actual value of the partition key, based on the keyName.
+    * If specified as "jobname" returns the jobname.
+    * If specified as "hostname" returns the hostname,
+    * else returns the actual value specified.
+    *
+    * @param keyName  the desired key name (hostname, jobname, none, or another custom name)
+    * @param jobName  the current job name
+    * @param hostname the current host name
+    */
+  def getMetricsReporterStreamPartitionKeyValue(keyName: String, jobName: String, hostname: String) = keyName match {
+    case "jobname" => jobName
+    case "hostname" => hostname
+    case _ => keyName
```

Comment:
Is this keyName or keyValue? The javadoc says "else returns the actual value specified", but the input argument only have key name. In addition, if "jobname" and "hostname" are special key names, I would prefer to define those as constant in MetricsConfig.